code/modules/vehicles/mecha/equipment/weapons/weapons.dm

```diff
@@ -457,7 +457,7 @@
 
 /obj/item/mecha_parts/mecha_equipment/weapon/ballistic/launcher/punching_glove/ui_act(action, list/params)
 	. = ..()
-	if(action == "toggle")
+	if(action == "change_mode")
```

Comment:
these should really be defines because the typo monster wants to eat our liver but no need to do that in this pr ofc

Reply:
this is a problem across all of ui_act, need a better general solution. i have a few things in mind